types/src/proof/position.rs

```diff
@@ -75,6 +75,11 @@ impl Position {
         treebits::level(self.0)
     }
 
+    // Compute the position given the level and the pos count on this level.
+    pub fn from_level_and_pos(level: u32, pos: u64) -> Self {
+        Self::from_inorder_index(treebits::node_from_level_and_pos(level, pos))
```

Comment:
That's what I'm thinking as well. I feel exposing both `position` and `treebits` is a bit confusing. That's why I added a wrapper here instead of using `treebits` directly. Can do a follow-up diff for that.